cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -97,6 +108,13 @@ DrawableSubmap::DrawableSubmap(const ::cartographer::mapping::SubmapId& id,
       "" /* title */, visible, "" /* description */, submap_category,
       SLOT(ToggleVisibility()), this);
   submap_node_->attachObject(manual_object_);
+
+  // TODO: Make it toggleable.
+  submap_id_text_.setTextAlignment(::rviz::MovableText::H_CENTER,
+                                   ::rviz::MovableText::V_ABOVE);
+  submap_id_text_node_->setPosition(ToOgre(kSubmapIdHeight));
```

Comment:
Maybe I misunderstand this, but can't you attach the `submap_id_text_` to the `scene_node_` (could you rebase? this is now called `submap_node_`) and use `submap_id_text_.setLocalTranslation()`?